### DISCUSSION TITLE
Transfer hidden tabs to normal tabs or unhide them in bulk

### DISCUSSION THREAD

**Discussion by mr-cheffy** [zen-browser/desktop#774] [Ideas]
### What feature would you like to see?

I migrated my tabs from firefox, once added, they are usually not loaded unless you click on it, in firefox, these tabs remain there until I open them, in Zen, and after closing the browser and open it again, it put all of these tabs in a 'hidden tabs' section instead of loading them into the normal one, I would like to keep them all in the default location or unhide them in bulk.
![Screenshot_20240825_034106](https://github.com/user-attachments/assets/6f9fca1b-7a39-4b00-8b21-93b8fccb0305)

 
 ~ _Created automatically from issue: https://github.com/zen-browser/desktop/issues/650_

**Comment by semenne** [comment]
I think this is pretty important...
I used Tab Session Manager to export my  600+ tabs from Firefox.
When importing them into Zen Browser, all the tabs are put in no workspace, so they all show as hidden.
This is because Zen Browser uses a Zen only  "zenWorkspace" property in the session store, to specify what workspace a tab belongs to.
There is no way to put the "hidden" imported tabs into the default workspace, except 1 by 1.
But even if I put everything in the right place, the problem is going to present itself again if I have a problem and have to restore an auto-saved backup session from Tab Session Manager: even if I have set the workspace for all the tabs, the custom "zenWorkspace" property is not saved, so the tabs will be all "hidden" again.
At the very least Zen Browser should always set a "default" workspace when a tab has none.

**Comment by HeroYui** [comment]
I've been a bit puzzled by the hidden tabs feature and wanted to share some thoughts on how it could be improved:

Transparency:
- It would be great to understand when and why tabs become hidden
- Could we get some dedicated settings for hidden tabs in the menu?

Usability Enhancements:
- An option to open a hidden tab when closing the last active one would be really helpful
- Multi-select for hidden tabs would be fantastic - makes it easier to bring them back

User Control:
- More options to manage the hiding/unhiding process would be appreciated

I think these changes would make the hidden tabs feature much more user-friendly and less confusing.
Thanks for considering my feedback!

**Comment by semenne** [comment]
For what I have seen, it seems hidden tabs are tabs that are not associated with any workspace.

Another thing I noticed is that if you right-click on a hidden tab and than click on select all, you will select all the tabs of the current workspace instead of all the hidden tabs. This could be even considered a bug.

**Comment by jedinjapan** [comment]
This remains an issue in the current release, and makes migrating complex sessions from Firefox quite difficult. The user cannot even simply re-create their sessions in Zen because the tabs are simply gone.
